README.md

```diff
@@ -1 +1,262 @@
+[zq README](https://github.com/brimsec/zq)
+to install the tools, which simply involves making sure you have Go installed etc,
+and running
+```
+make install
+```
+You also need ray installed, so do
+```
+pip install ray
+```
+
+## run the example
+
+To try out the example, run something like this:
+```
+./rayzardemo.py --find=":ip=192.168.0.51" --N=4 \
+        --filter="192.168.0.51 | count() by id.orig_h,id.resp_h" \
+        --merge="sum(count) as count by id.orig_h,id.resp_h" | zq -t -
+```
+Here, the `--find` argument specifies a `zar find` predicate,
+the `--filter` argument provides a zql command that is applied to each file found,
+and the `--merge` argument provides a zql command that combines the output of  
+each filtered result into a running result.  
+
+The degree of concurrency is controlled by the `--N` argument.
+
+In the toy example here, a zar search is implemented as a ray actor that traverses
+a zar archive enumerating the data files that match a provided search expression.
+This is implemented by the
+[ArchiveSearch class](https://github.com/brimsec/rayzar/blob/2b6efde844abc30eb60e3e700987d16de441febe/rayzardemo.py#L44).
+The `next` method on this actor returns each subsequent file matching the search
+at each call.  (This could be more efficiently streamed but currently all of the
+results are read in a blocking manner then they are return one-by-one via the
+`next` method.)
+
+For each matching file, a
+[Ray task](https://github.com/brimsec/rayzar/blob/2b6efde844abc30eb60e3e700987d16de441febe/rayzardemo.py#L30)
+is run to read the file by inefficiently execing zq, applying the specified zql filter,
+and returning the data return by zq in one shot.
+
+The [Aggregator class](https://github.com/brimsec/rayzar/blob/2b6efde844abc30eb60e3e700987d16de441febe/rayzardemo.py#L63)
+is a Ray actor that aggregates together each of the results that is read and
+filtered through the zq reader.
+
+Upon completion, the binary
+zng output of the merge aggregation is sent to standard out and in the command above,
+converted it to text zng via `zq -t`.
+
+This is all incredibly inefficient because all
+the data is passed through the plasma object store --- this is just meant as a first cut
+here to start brainstorming.  A better approach would be to have a data reader
+efficiently read the filtered results straight into memory that could then be
+operated upon before being resulted into the plasma object store in the Arrow format.
+
+## incremental deployment model
+
+Supposing you had an existing deployment of stored data that was accessed by Ray,
+you could conceivably index this existing data with zar and add a search capability
+to any jobs that traverse the stored data file by file using the zar-find actor
+from this example.  In fact, zar find could be parallelized by providing
+a different zar root to search for the set of files under that root that match.
```

Comment:
`In fact, zar find ...` - I would drop this sentence, as it describes a particular sub-optimization for 'zar find' that distracts from the overall goal of the paragraph to introduce the concept of using indexes. The next sentence (`Then, any existing code ...`) still works without it.